### PR TITLE
upload coverage to codecov.io

### LINKED ISSUE #50: Integrate with codecov.io
We are currently doing some [nasty things](https://github.com/veraison/go-cose/blob/remote-signing/.github/workflows/ci-go-cover.yml#L36) to report the coverage. Coverage results are not stored anywhere and one never knows how the coverage changed from on PR to another.

I propose to integrate with https://codecov.io, which is a great and free service for monitoring and reporting code coverage.

It is easy to integrate with GH actions and with Go build-in coverage tooling.

I can't do the integration myself because someone with admin rights needs to log into codecov and do the setup.

### PR DESCRIPTION
**Pull request by qmuntal**
Fix #50

Signed-off-by: qmuntal <qmuntaldiaz@microsoft.com>

### CODE REVIEW

**Review by thomas-fossati — APPROVED** [review]
awesome

**Review by SteveLasker — APPROVED** [review]
Thanks @qmuntal, 
LGTM

File: .github/workflows/ci-go-cover.yml [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-    - name: Checkout code
-      uses: actions/checkout@v3
-    - name: Install Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: 1.18.x
-    - name: Go Coverage
-      run: |
-        go version
-        go test -short -cover | grep "^.*coverage:.*of statements$" | python -c "import os,re,sys; cover_rpt = sys.stdin.read(); print(cover_rpt) if len(cover_rpt) != 0 and len(cover_rpt.splitlines()) == 1 else sys.exit(1); min_cover = float(re.findall(r'\d*\.\d+|\d+', os.environ['GITHUB_WORKFLOW'])[0]); cover = float(re.findall(r'\d*\.\d+|\d+', cover_rpt)[0]); sys.exit(1) if (cover > 100) or (cover < min_cover) else sys.exit(0)"
```
Comment (shizhMSFT) [inline .github/workflows/ci-go-cover.yml:36]: Should we retain the `cover >= 89%` check to ensure we always have coverage >= 89%? It seems codecov does not provide this functionality.

Comment (qmuntal) [reply .github/workflows/ci-go-cover.yml:36]: Good catch, forgot to do that. Codecov does support minimum coverage checks: https://docs.codecov.com/docs/codecovyml-reference#coveragerange. I'm trying to set it up.

Comment (qmuntal) [reply .github/workflows/ci-go-cover.yml:36]: Done! https://github.com/veraison/go-cose/pull/84/commits/6bb266d13c51f54d1f4266252fe8d5b911b197a8

**Pull request merged**